packages/crypto/packages/util/README.md

```diff
@@ -0,0 +1,110 @@
+# `@iroha/crypto-util`
+
+Utilities shared across crypto packages.
+
+## Usage
+
+```ts
+import { freeScope } from '@iroha2/crypto-util'
+```
+
+See [`@iroha2/crypto-core`](https://github.com/hyperledger/iroha-javascript/tree/iroha2/packages/crypto/packages/core) package for details.
+
+### Work with `Free`s
+
+All WASM objects have `.free()` method to trigger manual deallocation:
+
+```ts
+interface Free {
+  free: () => void
+}
+```
+
+#### Wrap everything with `FreeGuard`
+
+`FreeGuard` is a wrapper around any `Free` object which:
+
+- gives access to the inner object
+- throws a friendly error when the object is freed (more friendly than a Rust panic in WASM)
+- tracks itself in the global [`FREE_HEAP`](#inspect-active-guards-in-freeheap)
+- attaches itself to the current [scope](#scope-guards) if there is any
+
+```ts
+import { FreeGuard, freeScope, FREE_HEAP, Free } from '@iroha2/crypto-util'
+
+declare const wasmObject: Free & { do_stuff: () => void }
+
+// add this guard to the current scope
+const guard = new FreeGuard(wasmObject)
+
+// access the object
+guard.object.do_stuff()
+
+// call `.free()` on the object itself, clear tracks and empty the guard,
+// so it will be not usable anymore
```

Comment:
```suggestion
// call `.free()` on the object itself to clear tracks and empty the guard
// after you no longer need to use it
```

Reply:
Nope, the meaning is not the same.

This is what I want to say in an expanded form:

> `guard.free()` does:
> - call `.free()` on the guarded object itself
> - clear tracks, i.e. removes the *guard* from `FREE_HEAP` and from the scope if there is any
> - empty the guard, i.e. removes the internal object and makes it not usable anymore. It will throw error if the internal object is accessed with `guard.object`

Reply:
So let's just write exactly what you said:

```suggestion
// call `.free()` on the guarded object to:
// - clear tracks, i.e. remove the guard from `FREE_HEAP` and from the scope (if there is any)
// - empty the guard, i.e. remove the internal object and make it unusable
//   so that it could no longer be accessed with `guard.object`
```

Reply:
Okay )

Reply:
Done